pkg/openwsn/contrib/radio.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (C) 2018 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_openwsn
+ * @{
+ *
+ * @file
+ *
+ * @author  Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author  Oliver Hahm <oliver.hahm@inria.fr>
+ * @}
+ */
+#include <sys/uio.h>
+
+#include "leds.h"
+#include "debugpins.h"
+#include "sctimer.h"
+
+#include "openwsn.h"
+#include "radio_ow.h"
+
+#include "net/netopt.h"
+#include "net/ieee802154.h"
+#include "net/netdev/ieee802154.h"
+
+#define ENABLE_DEBUG                (0)
+#include "debug.h"
+
+openwsn_radio_t openwsn_radio;
+
+static void _event_cb(netdev_t *dev, netdev_event_t event);
+
+/* stores the NETDEV_EVENT_ISR capture time to tag the following NETDEV_EVENT */
+static PORT_TIMER_WIDTH _txrx_event_capture_time = 0;
+
+int openwsn_radio_init(netdev_t *netdev)
+{
+    assert(netdev);
+
+    DEBUG_PUTS("[OpenWSN radio]: initialize riot-adaptation");
+    openwsn_radio.dev = netdev;
+
+    if (netdev->driver->init(netdev)) {
+        DEBUG_PUTS("[OpenWSN radio]: unable to initialize device");
+        return -1;
+    }
+    netdev->event_callback = _event_cb;
+
+    DEBUG_PUTS("[OpenWSN radio]: put radio in standby");
+    netopt_state_t state = NETOPT_STATE_STANDBY;
+    netdev->driver->set(netdev, NETOPT_STATE, &(state), sizeof(state));
+
+    DEBUG_PUTS("[OpenWSN radio]: set needed netdev options");
+    netopt_enable_t enable;
+    /* Enable needed IRQs */
+    enable = NETOPT_ENABLE;
+    netdev->driver->set(netdev, NETOPT_RX_START_IRQ, &(enable), sizeof(enable));
+    enable = NETOPT_ENABLE;
+    netdev->driver->set(netdev, NETOPT_RX_END_IRQ, &(enable), sizeof(enable));
+    enable = NETOPT_ENABLE;
+    netdev->driver->set(netdev, NETOPT_TX_END_IRQ, &(enable), sizeof(enable));
+    enable = NETOPT_DISABLE;
+    /* Enable basic mode, no AUTOACK. no CSMA , no frame filtering */
+    netdev->driver->set(netdev, NETOPT_AUTOACK, &(enable), sizeof(enable));
+    enable = NETOPT_DISABLE;
+    netdev->driver->set(netdev, NETOPT_CSMA, &(enable), sizeof(enable));
+    enable = NETOPT_ENABLE;
+    netdev->driver->set(netdev, NETOPT_RAWMODE, &(enable), sizeof(enable));
+    uint8_t retrans = 0;
+    /* MAC layer will handle retransmissions */
+    netdev->driver->set(netdev, NETOPT_RETRANS, &(retrans), sizeof(uint8_t));
+    /* Sniffer mode only */
+    enable = NETOPT_ENABLE;
+    netdev->driver->set(netdev, NETOPT_PROMISCUOUSMODE, &(enable),
```

Comment:
Are you sure about that?

Reply:
I think I added this when adding support for `cc2538`, since OpenWSN did this for that platform. I'll take a second look.

Reply:
removed